src/index.tsx

```diff
@@ -17,6 +20,8 @@ ReactDOM.render(
       <BrowserRouter basename="/">
         <div>
           <Route path="/" component={App} />
+          <Route path="/myMatches" component={MatchesList} />
+          <Route path="/addMatches" component={AddMatches} />
```

Comment:
Follow the route structure as in GameBuilder: https://github.com/yoav-zibin/GameBuilder/blob/master/src/App.tsx

Also, use PrivateRoute (a component added by my earlier) instead of simple route.